src/wizards/newspack/views/settings/additional-brands/brand-upsert.tsx

```diff
@@ -0,0 +1,409 @@
+/**
+ * Additional Brands Brand page. Used to edit and create.
+ */
+
+/**
+ * WordPress dependencies.
+ */
+import { __ } from '@wordpress/i18n';
+import { addQueryArgs, cleanForSlug } from '@wordpress/url';
+import { Fragment, useState, useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies.
+ */
+import {
+	Router,
+	Card,
+	Grid,
+	Button,
+	SectionHeader,
+	TextControl,
+	ImageUpload,
+	ColorPicker,
+	SelectControl,
+	RadioControl,
+	hooks,
+} from '../../../../../components/src';
+
+import './style.scss';
+
+const { useParams } = Router;
+
+const {
+	themeColors: registeredThemeColors,
+	menuLocations,
+	menus: availableMenus,
+} = window.newspackSettings[ 'additional-brands' ].sections.additionalBrands;
+
+export default function Brand( {
+	brands = [],
+	upsertBrand,
+	wizardApiFetch,
+	fetchLogoAttachment,
+}: {
+	brands: Brand[];
+	editBrand?: number;
+	upsertBrand: ( brandId: number, brand: Brand ) => void;
+	wizardApiFetch: WizardApiFetch;
+	fetchLogoAttachment: ( brandId: number, logoId: number ) => void;
+} ) {
+	const { brandId = '0' } = useParams();
+	const selectedBrand = brands.find( ( { id } ) => id === Number( brandId ) );
+
+	const [ brand, updateBrand ] = hooks.useObjectState< Brand >( {
+		id: 0,
+		name: 'Brand Name',
+		slug: 'brand-slug',
```

Comment:
If unchanged by the user, this default value will not allow for the brand name to have an auto-generated slug. E.g.: Brand name becoming `brand-name`. This should be an input placeholder so it doesn't intefere with the form input values.

Reply:
Updated 3bd7e88c729bf50189c85b2fe5e580c2d22df6a9